packages/components/src/sandbox/index.native.js

```diff
@@ -0,0 +1,252 @@
+
+		// Put the html snippet into a html document, and update the state to refresh the WebView,
+		// we can use this in the future to inject custom styles or scripts.
+		// Scripts go into the body rather than the head, to support embedded content such as Instagram
+		// that expect the scripts to be part of the body.
+		const htmlDoc = (
+			<html lang={ lang } className={ type }>
+				<head>
+					<title>{ title }</title>
+					<meta
+						name="viewport"
+						content="width=device-width, initial-scale=1"
+					></meta>
+					<style dangerouslySetInnerHTML={ { __html: style } } />
+					{ styles.map( ( rules, i ) => (
+						<style
+							key={ i }
+							dangerouslySetInnerHTML={ { __html: rules } }
+						/>
+					) ) }
+				</head>
+				<body
+					data-resizable-iframe-connected="data-resizable-iframe-connected"
+					className={ type }
+				>
+					<div dangerouslySetInnerHTML={ { __html: html } } />
+					<script
+						type="text/javascript"
+						dangerouslySetInnerHTML={ {
+							__html: observeAndResizeJS,
+						} }
+					/>
+					{ scripts.map( ( src ) => (
+						<script key={ src } src={ src } />
+					) ) }
+				</body>
+			</html>
+		);
+		return '<!DOCTYPE html>' + renderToString( htmlDoc );
+	}
+
+	function updateContentHtml( forceRerender = false ) {
+		const newContentHtml = getHtmlDoc();
+
+		if ( forceRerender && contentHtml === newContentHtml ) {
+			// The re-render is forced by updating the state with empty HTML,
+			// waiting for the JS code to be executed with "setImmediate" and then
+			// setting the content HTML again.
+			setContentHtml( '' );
+			setImmediate( () => setContentHtml( newContentHtml ) );
+		} else {
+			setContentHtml( newContentHtml );
+		}
+	}
+
+	function checkMessageForResize( event ) {
+		// Attempt to parse the message data as JSON if passed as string
+		let data = event.nativeEvent.data || {};
+
+		if ( 'string' === typeof data ) {
+			try {
+				data = JSON.parse( data );
+			} catch ( e ) {}
+		}
+
+		// Update the state only if the message is formatted as we expect,
+		// i.e. as an object with a 'resize' action.
+		if ( 'resize' !== data.action ) {
+			return;
+		}
+
+		setWidth( data.width );
+		setHeight( data.height );
+	}
+
+	function getAspectRatio() {
+		const contentWidth = Math.ceil( width );
+		const contentHeight = Math.ceil( height );
+
+		if ( contentWidth && contentHeight ) {
+			return contentWidth / contentHeight;
+		}
+
+		return 1;
+	}
+
+	function onChangeDimensions( dimensions ) {
+		setIsLandscape( dimensions.window.width >= dimensions.window.height );
+	}
+
+	useEffect( () => {
+		Dimensions.addEventListener( 'change', onChangeDimensions );
+		return () => {
+			Dimensions.removeEventListener( 'change', onChangeDimensions );
+		};
+	}, [] );
+
+	useEffect( () => {
+		updateContentHtml();
+	}, [ html, title, type, styles, scripts ] );
+
+	return (
+		<WebView
+			key={ key }
+			ref={ ref }
+			source={ { baseUrl: providerUrl, html: contentHtml } }
```

Comment:
Seems like `baseUrl` is used for any relative links in the HTML. Maybe we can make the  `scripts` with relative paths work with this?

Reply:
Yeah, good point, I'll figure out a way to get the URL of the post because that value is more suitable for use it as the base URL.

Reply:
> Seems like baseUrl is used for any relative links in the HTML. Maybe we can make the scripts with relative paths work with this?

Yep, it's actually working that way, the problem is that it might not match the base URL with the expected for the scripts. As an example, referenced in [this comment](https://github.com/WordPress/gutenberg/pull/33426#discussion_r679982499), the script `/wp-content/js/devicepx.js` that is injected when embedding a GIF from Giphy, expects the base URL to be `https://wordpress.com` which doesn't match with the provider, although in this case, the value of `providerURL` is `undefined` not `giphy.com` 🤔 .

An alternative, as I commented [here](https://github.com/WordPress/gutenberg/pull/33426#discussion_r682684952), could be to use `https://wordpress.com` as a fallback when provider URL is not defined.

Reply:
As posted in [this comment](https://github.com/WordPress/gutenberg/pull/33426#discussion_r684317741), for now, we're not going to inject the scripts.